README.md

```diff
@@ -401,6 +405,98 @@ If the modification of the Application resource is completed successfully, the `
+[volumemount]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#volumes
+
+### Resource Type Schema
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: ClusterApplicationResourceMapping
+metadata:
+  name:                 # string
+  generation:           # int64, defined by the Kubernetes control plane
+  ...
+spec:
+  versions:             # []Version
+  - version:            # string
+    containers:         # []string, optional
+    envs:               # []string, optional
+    volumeMounts:       # []string, optional
+    volumes:            # string
+```
+
+### Container-based Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: ClusterApplicationResourceMapping
+metadata:
+ name:  cronjobs.batch
+spec:
+  versions:
+  - version: v1beta1
+    containers:
+    - .spec.jobTemplate.spec.template.spec.containers
+    - .spec.jobTemplate.spec.template.spec.initContainers
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+  - version: v2alpha1
+    containers:
+    - .spec.jobTemplate.spec.template.spec.containers
+    - .spec.jobTemplate.spec.template.spec.initContainers
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+```
+
+### Element-based Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: ClusterApplicationResourceMapping
+metadata:
+ name:  cronjobs.batch
+spec:
+  versions:
+  - version: v1beta1
+    env:
+    - .spec.jobTemplate.spec.template.spec.containers[*].env
+    - .spec.jobTemplate.spec.template.spec.initContainers[*].env
+    volumeMounts:
+    - .spec.jobTemplate.spec.template.spec.containers[*].volumeMounts
+    - .spec.jobTemplate.spec.template.spec.initContainers[*].volumeMounts
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+  - version: v2alpha1
+    env:
+    - .spec.jobTemplate.spec.template.spec.containers[*].env
+    - .spec.jobTemplate.spec.template.spec.initContainers[*].env
+    volumeMounts:
+    - .spec.jobTemplate.spec.template.spec.containers[*].volumeMounts
+    - .spec.jobTemplate.spec.template.spec.initContainers[*].volumeMounts
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+```
+
+### Reconciler Implementation
+
+A reconciler implementation that supports `ClusterApplicationResourceMapping`s **MUST** support `ServiceBinding` resources that refer to applications that are not PodSpec-able.  If no Application Resource Mapping exists for the `ServiceBinding` application resource type, the reconciliation **MUST** fail.
+
+If a `ClusterApplicationResourceMapping` defines `containers`, the reconciler **MUST** first resolve a set of candidate locations in the application resource addressed by the `ServiceBinding` using the `Container` type (`.env`, `.volumeMounts`) for all available containers and then filter that collection by the `ServiceBinding` `.spec.application.containers` filter before applying the appropriate modification.
```

Comment:
Since we still allow binding to containers by name or index, we should clarify how index based matching occurs. I propose that index based matching only apply to the first entry in the ClusterApplicationResourceMapping's containers array, and name matching apply to all.